homeassistant/components/lifx/coordinator.py

```diff
@@ -183,3 +211,136 @@ async def async_set_color_zones(
                 apply=apply,
             )
         )
+
+    async def async_identify_bulb(self) -> None:
+        """Identify the device by flashing it three times."""
+        if self.device.power_level:
+            # just flash the bulb for three seconds
+            await self.async_set_waveform_optional(value=IDENTIFY_WAVEFORM)
+            return
+        # Turn the bulb on first, flash for 3 seconds, then turn off
+        await self.async_set_power(state=True, duration=1)
+        await self.async_set_waveform_optional(value=IDENTIFY_WAVEFORM)
+        await asyncio.sleep(3)
+        await self.async_set_power(state=False, duration=1)
+
+
+class LIFXSensorUpdateCoordinator(LIFXUpdateCoordinator):
+    """DataUpdateCoordinator for sensor data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        connection: LIFXConnection,
+    ) -> None:
+        """Initialize LIFX DataUpdateCoordinator."""
+        super().__init__(
+            hass,
+            entry,
+            connection,
+            timedelta(seconds=60),
+            Debouncer(hass, _LOGGER, cooldown=0, immediate=True),
+        )
+
+        self.update_rssi: bool = False
+        self.update_uptime: bool = False
+        self.update_hev_cycle_duration: bool = False
+        self.update_hev_cycle_remaining: bool = False
+        self.update_hev_cycle_last_power: bool = False
+        self.update_last_hev_cycle_result: bool = False
+
+        self.rssi: int = 0
+        self.uptime: int = 0
+        self.hev_cycle_duration: int = 7200
+        self.hev_cycle_remaining: int = 0
+        self.hev_cycle_last_power: bool = False
+        self.last_hev_cycle_result: str = STATE_UNKNOWN
+
+    def get_rssi_unit_of_measurement(self) -> str | None:
+        """Return the unit of measurement for the RSSI sensor."""
+        if AwesomeVersion(self.device.host_firmware_version) > RSSI_DBM_FW:
+            return SIGNAL_STRENGTH_DECIBELS_MILLIWATT
+        return SIGNAL_STRENGTH_DECIBELS
+
+    async def _async_update_data(self) -> None:
+        """Update sensor data for enabled sensors."""
+
+        async def _async_update_uptime() -> None:
+            """Fetch the current uptime value."""
+            try:
+                uptime = get_uptime_from_hostinfo(
+                    await async_execute_lifx(self.device.get_hostinfo)
+                )
+                if uptime is not None:
+                    self.uptime = uptime
+
+            except asyncio.TimeoutError as ex:
+                raise UpdateFailed(
+                    f"Failed to fetch uptime: {self.device.label} ({self.device.ip_addr})"
+                ) from ex
```

Comment:
This could be a decorator instead since you are doing the same thing multiple places